src/lib.rs

```diff
@@ -464,6 +464,50 @@ pub fn sorensen_dice(a: &str, b: &str) -> f64 {
     (2 * intersection_size) as f64 / (a.len() + b.len() - 2) as f64
 }
 
+/// Uses LCS algorithm to find longest common subsequence
+/// and then divides it by the length of the longes string
+/// ```
+/// use strsim::lcs_normalized;
+///
+/// assert_eq!(1.0, lcs_normalized("", ""));
+/// assert_eq!(0.0, lcs_normalized("", "umbrella"));
+/// assert_eq!(0.8, lcs_normalized("night", "fight"));
+/// assert_eq!(1.0, lcs_normalized("ferris", "ferris"));
+/// ```
+pub fn lcs_normalized(left: impl AsRef<str>, right: impl AsRef<str>) -> f64 {
+    let (len1, len2) = (left.as_ref().len(), right.as_ref().len());
+    let lcs_len = lcs_length(left.as_ref(), right.as_ref());
+    let size = max(len1, len2);
+    // Empty strings should match
+    if size == 0 { 1.0 } else { lcs_len as f64 / size as f64 }
+}
+
+#[inline]
+fn get_shorter_longer_strings(left: impl AsRef<str>, right: impl AsRef<str>) -> (String, String) {
+    if left.as_ref().len() < right.as_ref().len() {
+        (left.as_ref().to_string(), right.as_ref().to_string())
+    } else {
+        (right.as_ref().to_string(), left.as_ref().to_string())
+    }
+}
+
+#[inline]
+fn lcs_length(left: impl AsRef<str>, right: impl AsRef<str>) -> usize {
+    let (left, right) = get_shorter_longer_strings(left, right);
+    let mut table = vec![vec![0 as usize; left.len() + 1]; 2];
+    for rletter in right.chars() {
+        for (col, lletter) in left.chars().enumerate() {
+            if rletter == lletter {
+                table[1][col + 1] = 1 + table[0][col];
+            } else {
+                table[1][col + 1] = max(table[0][col + 1], table[1][col]);
+            }
```

Comment:
In rust I would probably use something like:
```suggestion
           table[1][col + 1] =  if rletter == lletter {
                1 + table[0][col]
            } else {
               max(table[0][col + 1], table[1][col])
            };
```
instead.